compiler/passes/resolveIntents.cpp

```diff
@@ -97,7 +97,11 @@ IntentTag blankIntentForType(Type* t) {
 
 IntentTag concreteIntent(IntentTag existingIntent, Type* t) {
   if (existingIntent == INTENT_BLANK) {
-    return blankIntentForType(t);
+    if (t->symbol->hasFlag(FLAG_REF)) {
+      return INTENT_REF;
+    } else {
+      return blankIntentForType(t);
+    }
```

Comment:
Please add a comment explaining why returning INTENT_REF for ref types. Is this a convention? When do we call concreteIntent() on ref types?